### PR TITLE
#1740 #2003 #2059 #2090 #2092 #2098 

### PR DESCRIPTION
**Pull request by MikhailArkhipov**

### CODE REVIEW

File: src/R/Editor/Test/Utility/FunctionIndexBasedTest.cs [hunk 1]
```diff
@@ -25,8 +25,8 @@ public abstract class FunctionIndexBasedTest : IAsyncLifetime {
             _functionIndex = _exportProvider.GetExportedValue<IFunctionIndex>();
         }
 
-        public Task InitializeAsync() {
-            return _packageIndex.InitializeAsync(_functionIndex);
+        public async Task InitializeAsync() {
+            await _packageIndex.InitializeAsync(_functionIndex);
```
Comment (AlexanderSher) [inline src/R/Editor/Test/Utility/FunctionIndexBasedTest.cs]: Why?

File: src/Languages/Editor/Impl/Extensions/TextViewExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.Languages.Editor.Shell;
+using Microsoft.VisualStudio.Language.Intellisense;
+using Microsoft.VisualStudio.Text.Editor;
+
+namespace Microsoft.Languages.Editor.Extensions {
+    public static class TextViewExtensions {
+        public static bool IsAutoInsertAllowed(ITextView textView) {
```
Comment (AlexanderSher) [inline src/Languages/Editor/Impl/Extensions/TextViewExtensions.cs]: Do you want to make this method an extension?

**Comment by AlexanderSher** [comment]
:shipit:

File: src/R/Editor/Test/Utility/FunctionIndexBasedTest.cs (hunk 1 above)
Comment (MikhailArkhipov) [inline src/R/Editor/Test/Utility/FunctionIndexBasedTest.cs]: Mostly for consistency with Dispose, but I'll revert it

**Pull request merged**